### PR TITLE
Update code for photutils 1.1.0 

### PR DESCRIPTION
**Pull request by larrybradley**
This PR removes old code for older versions of `photutils` and updates the code for the latest `photutils` release (version 1.1.0).

### CODE REVIEW

File: jwst/source_catalog/source_catalog.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 """
+from photutils.segmentation import detect_threshold
```
Comment (jdavies-st) [inline jwst/source_catalog/source_catalog.py:2]: This probably shouldn't be here.

**Review by jdavies-st — COMMENTED** [review]
When I run one of our NIRCam imaging regression tests

```
$ pytest --bigdata --basetemp=/tmp/scratch jwst/regtest/test_nircam_image.py
```

I get the following error:

```pytb
>       Step.from_cmdline(args)

/Users/jdavies/dev/jwst/jwst/regtest/test_nircam_image.py:74: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/stpipe/step.py:173: in from_cmdline
    return cmdline.step_from_cmdline(args)
/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/stpipe/cmdline.py:339: in step_from_cmdline
    step.run(*positional)
/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/stpipe/step.py:407: in run
    step_result = self.process(*args)
/Users/jdavies/dev/jwst/jwst/pipeline/calwebb_image3.py:85: in process
    input_models = self.tweakreg(input_models)
/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/stpipe/step.py:407: in run
    step_result = self.process(*args)
/Users/jdavies/dev/jwst/jwst/tweakreg/tweakreg_step.py:75: in process
    catalog = make_tweakreg_catalog(
/Users/jdavies/dev/jwst/jwst/tweakreg/tweakreg_catalog.py:83: in make_tweakreg_catalog
    catalog = sources[columns]
/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/astropy/table/table.py:1646: in __getitem__
    out = self.__class__([self[x] for x in item],
/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/astropy/table/table.py:1646: in <listcomp>
    out = self.__class__([self[x] for x in item],
/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/astropy/table/table.py:1640: in __getitem__
    return self.columns[item]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <TableColumns names=('id','xcentroid','ycentroid','sharpness','roundness1','roundness2','npix','sky','peak','flux','mag')>
item = 'label'

    def __getitem__(self, item):
        """Get items from a TableColumns object.
        ::
    
          tc = TableColumns(cols=[Column(name='a'), Column(name='b'), Column(name='c')])
          tc['a']  # Column('a')
          tc[1] # Column('b')
          tc['a', 'b'] # <TableColumns names=('a', 'b')>
          tc[1:3] # <TableColumns names=('b', 'c')>
        """
        if isinstance(item, str):
>           return OrderedDict.__getitem__(self, item)
E           KeyError: 'label'

/Users/jdavies/miniconda3/envs/jwst/lib/python3.9/site-packages/astropy/table/table.py:239: KeyError
```

File: jwst/source_catalog/source_catalog.py (hunk 1 above)
Comment (larrybradley) [reply jwst/source_catalog/source_catalog.py]: Oops.  It's removed.

**Comment by larrybradley** [comment]
@jdavies-st Thanks for catching that error. It should be resolved now.

**Review by jdavies-st — APPROVED** [review]
Looks good.  I think the drop in coverage will increase back up once the night's regression tests run.  👍

**Pull request merged**